src/github_runner_image_builder/cli.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Main entrypoint for github-runner-image-builder cli application."""
+import argparse
+import itertools
+import sys
+from pathlib import Path
+from typing import Literal, cast
+
+from github_runner_image_builder import builder
+from github_runner_image_builder.builder import BuildImageConfig
+from github_runner_image_builder.config import (
+    LTS_IMAGE_VERSION_TAG_MAP,
+    BaseImage,
+    get_supported_arch,
+)
+
+
+# This is a class used for type hinting argparse.
+class ActionsNamespace(argparse.Namespace):  # pylint: disable=too-few-public-methods
+    """Action positional argument namespace.
+
+    Attributes:
+        action: CLI action positional argument.
+        base: The base image to build.
+        output: Path of the output image file.
+    """
+
+    action: Literal["install", "build"]
+    base: Literal["22.04", "jammy", "24.04", "noble"]
```

Comment:
Used here in an `BaseImage` like architecture these should be close to each other to simplify addition on new bases

Reply:
Moved to config module, thank you!